### PR TITLE
Feat : Add redirect button to SelectAsync Comp if options is empty

### PR DESCRIPTION
**Pull request by parteekcoder**
## Description

If list is empty 
       it will show add Payment . On clicking it user will be redirected to add payment mode page. 
else 
       it will show only payment modes. on clicking that modes , user can select them
## Related Issues

#496 

## Checklist

- [x] I have tested these changes
- [ ] I have updated the relevant documentation
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the codebase
- [x] My changes generate no new warnings or errors
- [x] The title of my pull request is clear and descriptive

### CODE REVIEW

File: frontend/src/components/SelectAsync/index.jsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React, { useState, useEffect, useRef } from 'react';
 import { request } from '@/request';
 import useFetch from '@/hooks/useFetch';
 import { Select } from 'antd';
+import { useHistory } from 'react-router-dom/cjs/react-router-dom.min';
 
 export default function SelectAsync({
   entity,
```
Comment (salahlalami) [inline frontend/src/components/SelectAsync/index.jsx:8]: add param 'withRedirect' default is false ,
add param 'urlToRedirect' default is '/' ,

File: frontend/src/components/SelectAsync/index.jsx [hunk 2]
```diff
@@ -34,18 +37,30 @@ export default function SelectAsync({
     }
   }, [value]);
 
+  const handleSelectChange = (newValue) => {
+    if (newValue === 'addPayment') {
```
Comment (salahlalami) [inline frontend/src/components/SelectAsync/index.jsx:41]: if (newValue === 'addPayment' && withRedirect) {

File: frontend/src/components/SelectAsync/index.jsx [hunk 3]
```diff
@@ -34,18 +37,30 @@ export default function SelectAsync({
...
+      // Navigate to another page when "Add payment" is selected
+      history.push('/payment/mode');
```
Comment (salahlalami) [inline frontend/src/components/SelectAsync/index.jsx:43]: history.push(urlToRedirect);

File: frontend/src/components/SelectAsync/index.jsx [hunk 4]
```diff
@@ -34,18 +37,30 @@ export default function SelectAsync({
...
       value={currentValue}
-      onChange={(newValue) => {
-        // setCurrentValue(newValue[outputValue] || newValue);
-        if (onChange) {
-          onChange(newValue[outputValue] || newValue);
-        }
-      }}
+      onChange={handleSelectChange}
     >
+      {selectOptions.length === 0 && (
```
Comment (salahlalami) [inline frontend/src/components/SelectAsync/index.jsx:59]: add :  withRedirect && selectOptions.length === 0

**Review by salahlalami — CHANGES_REQUESTED** [review]
this is core component that we can use it every where , it should be generic , and configurable

File: frontend/src/components/SelectAsync/index.jsx (hunk 1 above)
Comment (parteekcoder) [reply frontend/src/components/SelectAsync/index.jsx:8]: done

File: frontend/src/components/SelectAsync/index.jsx [hunk 5]
```diff
@@ -2,18 +2,24 @@ import React, { useState, useEffect, useRef } from 'react';
...
 import { Select } from 'antd';
+import { useHistory } from 'react-router-dom/cjs/react-router-dom.min';
 
 export default function SelectAsync({
   entity,
   displayLabels = ['name'],
   outputValue = '_id',
   value,
   onChange,
+  labelText = '',
```
Comment (salahlalami) [inline frontend/src/components/SelectAsync/index.jsx:13]: @parteekcoder  change it to redirectLabel or something similar

File: frontend/src/components/SelectAsync/index.jsx [hunk 6]
```diff
@@ -34,18 +40,30 @@ export default function SelectAsync({
...
-        // setCurrentValue(newValue[outputValue] || newValue);
-        if (onChange) {
-          onChange(newValue[outputValue] || newValue);
-        }
-      }}
+      onChange={handleSelectChange}
     >
+      {selectOptions.length === 0 && withRedirect && (
+        <Select.Option key="redirectURL" value="redirectURL">
+          {labelText}
```
Comment (salahlalami) [inline frontend/src/components/SelectAsync/index.jsx:64]: could you add inside select button with icon (PlusCircleOutlined ) + labelText .

check antd 4.X docs

**Review by salahlalami — CHANGES_REQUESTED** [review]

File: frontend/src/components/SelectAsync/index.jsx (hunk 6 above)
Comment (parteekcoder) [reply frontend/src/components/SelectAsync/index.jsx:64]: image

added

File: frontend/src/components/SelectAsync/index.jsx (hunk 5 above)
Comment (parteekcoder) [reply frontend/src/components/SelectAsync/index.jsx]: done

**Review by salahlalami — APPROVED** [review]

**Review by polymahh — APPROVED** [review]
LGTM 👍

**Pull request merged**